data_safe_haven/infrastructure/programs/sre/data.py

```diff
@@ -456,6 +457,154 @@ def __init__(
+            account_name=alphanumeric(
+                f"{''.join(truncate_tokens(stack_name.split('-'), 11))}desiredstate{sha256hash(self._name)}"
+            )[:24],
+            enable_https_traffic_only=True,
+            enable_nfs_v3=True,
+            encryption=storage.EncryptionArgs(
+                key_source=storage.KeySource.MICROSOFT_STORAGE,
+                services=storage.EncryptionServicesArgs(
+                    blob=storage.EncryptionServiceArgs(
+                        enabled=True, key_type=storage.KeyType.ACCOUNT
+                    ),
+                    file=storage.EncryptionServiceArgs(
+                        enabled=True, key_type=storage.KeyType.ACCOUNT
+                    ),
+                ),
+            ),
+            kind=storage.Kind.BLOCK_BLOB_STORAGE,
+            is_hns_enabled=True,
+            location=props.location,
+            network_rule_set=storage.NetworkRuleSetArgs(
+                bypass=storage.Bypass.AZURE_SERVICES,
+                default_action=storage.DefaultAction.DENY,
+                ip_rules=Output.from_input(props.data_configuration_ip_addresses).apply(
+                    lambda ip_ranges: [
+                        storage.IPRuleArgs(
+                            action=storage.Action.ALLOW,
+                            i_p_address_or_range=str(ip_address),
+                        )
+                        for ip_range in sorted(ip_ranges)
+                        for ip_address in AzureIPv4Range.from_cidr(ip_range).all_ips()
+                    ]
+                ),
+                virtual_network_rules=[
+                    storage.VirtualNetworkRuleArgs(
+                        virtual_network_resource_id=props.subnet_data_configuration_id,
+                    )
+                ],
+            ),
+            resource_group_name=resource_group.name,
+            sku=storage.SkuArgs(name=storage.SkuName.PREMIUM_ZRS),
+            opts=child_opts,
+            tags=child_tags,
+        )
+        # Deploy desired state share
+        container_desired_state = storage.BlobContainer(
+            f"{storage_account_data_configuration._name}_desired_state",
+            account_name=storage_account_data_desired_state.name,
+            container_name="desiredstate",
+            default_encryption_scope="$account-encryption-key",
+            deny_encryption_scope_override=False,
+            public_access=storage.PublicAccess.NONE,
+            resource_group_name=resource_group.name,
+            opts=ResourceOptions.merge(
+                child_opts,
+                ResourceOptions(parent=storage_account_data_desired_state),
+            ),
+        )
+        # Set storage container ACLs
+        BlobContainerAcl(
+            f"{container_desired_state._name}_acl",
+            BlobContainerAclProps(
+                acl_user="rwx",
+                acl_group="r-x",
+                acl_other="r-x",
```

Comment:
Why do we want these permissions?

Reply:
We probably actually want something like 550.

rx is needed for root to read and browse. Others shouldn't need it though.